src/2025h1/seamless-rust-cpp.md

```diff
@@ -50,29 +50,31 @@ For C/C++ this has led to an improvement along the technical front in terms of a
 
-Rust adoption is feasible in the following situations:
-
-##### Greenfield projects
+##### Feasible: New codebases with Rust-only dependencies
 
 This includes completely new projects as well as complete rewrites of existing projects, when such rewrites are socially and economically viable.
 
-##### Interprocess boundaries
+##### Feasible: Interprocess boundaries
 
-Projects with a natural interprocess boundary are easily migrated incrementally to Rust. Microservice architectures with their RPC/HTTP boundaries are one example of this.
+Projects with a natural interprocess boundary between components are more easily migrated to Rust. Because of the loose coupling enforced by the boundary, the project can be incrementally migrated one component at a time. Microservice architectures with their RPC/HTTP boundaries are one example of this.
 
-##### Small, simple API surface
+##### Feasible: Small, simple intraprocess API surface
 
 Projects with a small, simple API surface that can be manually expressed in terms of the C ABI. This boundary, expressed and invoked in `unsafe` code, is prone to human error. It can be maintainable when the surface is small enough, but this also means that Rust adoption can *decrease* safety at the language boundary.
 
-##### Larger API surfaces with limited vocabulary
+##### Feasible: Larger intraprocess API surface, but with limited vocabulary
 
 Projects with a limited API vocabulary are able to use one of the existing interop tools like bindgen, cbindgen, or cxx.
 
-The fact that all of these options exist and undergo active development is a testament to the value developers see in Rust adoption. However, they leave out a large portion of production code in use today: Projects that make rich use of a language such as C++ where comparatively limited interop support exists for Rust, and that link in enough code to make rewriting infeasible.
+##### Infeasible: Everything else
+
+The fact that all of these options exist and undergo active development is a testament to the value developers see in Rust adoption. However, they leave out a large portion of production use cases today: Projects that make rich use of an API in a language like C++ where comparatively limited interop support exists for Rust, and that link in enough code to make rewriting infeasible.
+
+Furthermore, the limitations of current interop tooling are not simply a matter of adding features. Many of them stem from a mismatch in the expressiveness of the two languages along various axes. As one example, C++ and Java both support overloading while Rust does not. In some cases this mismatch is broadly accepted as a missing feature in Rust that will be added in time. In others, Rust's lack of expressiveness may be considered a feature in itself.
```

Comment:
Probably worth calling out a few of these differences:
- Rust's different approach to inheritance (or perhaps the orientation towards composition)
- Generics vs templates
- Value semantics and default relocatability